java-manta-cli/src/main/java/com/joyent/manta/cli/MantaCLI.java

```diff
@@ -17,364 +19,367 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.security.NoSuchAlgorithmException;
+import java.util.List;
 import java.util.stream.Stream;
 import javax.crypto.SecretKey;
 import javax.crypto.spec.SecretKeySpec;
 
-/**
- * Class providing a CLI interface to the Java Manta SDK.
- *
- * @author <a href="https://github.com/dekobon">Elijah Zupancic</a>
- * @since 3.0.0
- */
+@CommandLine.Command(name = "java-manta-cli", sortOptions = false,
+        header = {
+             "@|cyan                 .     .             |@",
+             "@|cyan                 |_.-._|             |@",
+             "@|cyan               ./       \\.          |@",
+             "@|cyan          _.-'`           `'-._      |@",
+             "@|cyan       .-'        Java         '-.   |@",
+             "@|cyan     ,'_.._       Manta       _.._', |@",
+             "@|cyan     '`    `'-.           .-'`    `' |@",
+             "@|cyan               '.       .'           |@",
+             "@|cyan                 \\_/|\\_/           |@",
+             "@|cyan                    |                |@",
+             "@|cyan                    |                |@",
+             "@|cyan                    |                |@",
+             ""},
+        description = {
+                "",
+                "Basic manta cli commands using the java-client. Not a stable interface.", },
+        optionListHeading = "@|bold %nOptions|@:%n",
+         subcommands = {
+             MantaCLI.ConnectTest.class,
+             MantaCLI.DumpConfig.class,
+             MantaCLI.GenerateKey.class,
+             MantaCLI.ListDir.class,
+             MantaCLI.GetFile.class,
+             MantaCLI.PutFile.class,
+             MantaCLI.ValidateKey.class
+         })
+// Documented through CLI annotations
+@SuppressWarnings({"checkstyle:javadocmethod", "checkstyle:javadoctype", "checkstyle:javadocvariable"})
 public final class MantaCLI {
-    /**
-     * System dependent line-break.
-     */
-    private static final String BR = System.lineSeparator();
+    @CommandLine.Option(names = {"-v", "--version"}, help = true)
+    private boolean isVersionRequested;
 
-    /**
-     * Indent spaces to sub elements.
-     */
-    private static final String INDENT = "  ";
+    @CommandLine.Option(names = {"-h", "--help"}, help = true)
+    private boolean isHelpRequested;
 
-    /**
-     * Private constructor for class only containing static methods.
-     */
-    private MantaCLI() {
-    }
+    private MantaCLI() { }
 
-    /**
-     * Entrance to the CLI.
-     * @param argv parameters passed.
-     */
-    public static void main(final String[] argv) {
-        if (ArrayUtils.isEmpty(argv) || StringUtils.isBlank(argv[0])) {
-            System.out.println(help());
+    public static void main(final String[] args) {
+        final CommandLine application = new CommandLine(new MantaCLI());
+        application.registerConverter(Path.class, s -> Paths.get(s));
```

Comment:
nit: this could be written as

`application.registerConverter(Path.class, Paths::get);`